ome/omero-insight
# Using buildsrc and javapackager

This PR, essentially, makes insight _distributable_ and cleans up logic from the build.gradle file.

### Test 1
Test the application be run from gradle as Inisght and Importer

1. `gradle run` - runs insight
2. `gradle runImporter` - runs importer

### Test 2
Test an ImageJ jar is built 

1. `gradle imageJJar` creates a jar and outputs it to directory `build/libs`
2  Unzip the jar and check the _MANIFEST.MF_ (inside _META-INF_ folder) contains main-class org.openmicroscopy.shoola.MainIJPlugin

Edit: J-M
1. Unzip ``omero-insight-imagej-5.5.0-SNAPSHOT.zip`` under ``build/distributions`` and move the folder in the plugins folder of Fiji. Check that you can connect.

### Test 3
Create an installer using javapackager

1. ensure you are using Oracle JDK 1.8 and $JAVA_HOME path
      - run `javapackager` in cmd/terminal, it will fail if something isn't right
2. `gradle packageApplicationDmg` (if on Mac) or `gradle packageApplicationExe` if on Windows

Extras:
Windows openjdk with javapackager https://github.com/ojdkbuild/ojdkbuild/releases/download/1.8.0.201-1/java-1.8.0-openjdk-1.8.0.201-1.b09.ojdkbuild.windows.x86_64.msi
